File: 5.prototype/prototype.py
import copy

# Prototipo
class Circle:
    def __init__(self, radius):
        self.radius = radius

    def clone(self):
        return copy.deepcopy(self)

# Uso
original = Circle(5)
clone = original.clone()
clone.radius = 10

print("Original:", vars(original))
print("Clon:", vars(clone))

File: 05.prototype/prototype.ts
// Prototipo
class Circle {
    constructor(public radius: number) {}

    public clone(): Circle {
        return new Circle(this.radius);
    }
}

// Uso
const original = new Circle(5);
const clone = original.clone();
clone.radius = 10;

console.log("Original:", original);
console.log("Clon:", clone);